qa/common/check_copyright.py

```diff
@@ -144,40 +148,55 @@ def visit(path):
             prefix = '# '
         elif line.startswith('// '):
             prefix = '// '
-        elif not line.startswith(COPYRIGHT_YEAR_RE0[0]):
+        elif not line.startswith(COPYRIGHT_YEAR_RE[0]):
             print(
                 "incorrect prefix for copyright line, allowed prefixes '# ' or '// ', for "
                 + path + ": " + line)
             return False
 
-        start_year = 0
-        end_year = 0
+        # Check if the copyright year line matches the regex
+        # and see if the year(s) are reasonable
+        years = []
 
-        m = single_re.match(line[len(prefix):])
-        if m and len(m.groups()) == 1:
-            start_year = end_year = int(m.group(1))
+        copyright_row = line[len(prefix):]
+        if copyright_year_re.match(copyright_row):
+            for year in copyright_row.split("(c) " if "(c) " in
+                                            copyright_row else "Copyright "
+                                           )[1].split(" NVIDIA ")[0].split(","):
+                if len(year) == 4:  # 2021
+                    years.append(int(year))
+                elif len(year) == 2:  # 21
+                    years.append(int(year) + 2000)
+                elif len(year) == 9:  # 2021-2022
+                    years.append(int(year[0:4]))
+                    years.append(int(year[5:9]))
+                elif len(year) == 7 and year[4] == "-":  # 2021-22
+                    years.append(int(year[0:4]))
+                    years.append(int(year[5:7]) + 2000)
+                elif len(year) == 7 and year[2] == "-":  # 21-2022
```

Comment:
To that note we can modify the regex to flag this off instead of having to validate it afterwards.
By changing `(,(20)?[2-9][0-9](-(20)?[2-9][0-9])?)*,? N`
To `(,((20[2-9][0-9](-(20)?[2-9][0-9])?)|([2-9][0-9](-[2-9][0-9])?)))*,? N`

Reply:
Moved into regex.